src/front/glsl/parser/declarations.rs

```diff
@@ -190,22 +190,14 @@ impl<'source> ParsingContext<'source> {
                 })
                 .transpose()?;
 
-            // If the declaration has an initializer try to make a constant out of it,
-            // this is only strictly needed for global constant declarations (and if the
-            // initializer can't be made a constant it should throw an error) but we also
-            // try to do it for all other types of declarations.
-            let maybe_constant = if let Some((root, meta)) = init {
-                let is_const = ctx.qualifiers.storage.0 == StorageQualifier::Const;
-
-                match parser.solve_constant(ctx.ctx, root, meta) {
-                    Ok(res) => Some(res),
-                    // If the declaration is external (global scope) and is constant qualified
-                    // then the initializer must be a constant expression
-                    Err(err) if ctx.external && is_const => return Err(err),
-                    _ => None,
+            // If the variable is global and const qualified, solve the initializer for a constant
+            // and use that as the variable's initial value.
+            let is_const = ctx.qualifiers.storage.0 == StorageQualifier::Const;
+            let maybe_constant = match (init, ctx.external, is_const) {
```

Comment:
This will prevent non constant globals from being evaluated, try this instead:
```rust
let maybe_constant = if ctx.external {
  match parser.solve_constant(ctx.ctx, root, meta) {
      Ok(res) => Some(res),
      // If the declaration is external (global scope) and is constant qualified
      // then the initializer must be a constant expression
      Err(err) is_const => return Err(err),
      _ => None,
  }
} else {
  None
};
```

Reply:
Ah, thanks! There's a lot of cases here and I still didn't have it clear in my head.

Reply:
Actually, if we have a non-constant qualified global with an initializer that can't be solved, should that also be an error?

Reply:
No glsl allows globals to refer to other globals which can't be constified

Reply:
Fascinating, ok! I will go with this. Thanks again!